docs/configuration.html

```diff
@@ -308,6 +308,157 @@ <h3 class="anchor-heading"><a id="tieredstorageconfigs" class="anchor-link"></a>
+  </p>
+  
+  <p>You have the following options:</p>
+  <ul>
+    <li>
+      Use a custom provider by creating a class implementing the
+      <a href="/{{version}}/javadoc/org/apache/kafka/common/config/provider/ConfigProvider.html"><code>ConfigProvider</code></a> interface and packaging it into a JAR file.
+    </li>
+    <li>Use a built-in provider:</li>
+    <ul>
+      <li><a href="/{{version}}/javadoc/org/apache/kafka/common/config/provider/DirectoryConfigProvider.html"><code>DirectoryConfigProvider</code></a></li>
+      <li><a href="/{{version}}/javadoc/org/apache/kafka/common/config/provider/EnvVarConfigProvider.html"><code>EnvVarConfigProvider</code></a></li>
+      <li><a href="/{{version}}/javadoc/org/apache/kafka/common/config/provider/FileConfigProvider.html"><code>FileConfigProvider</code></a></li>
+    </ul>
+  </ul>
+  
+  <p>
+    To use a configuration provider, specify it in your configuration using the <code>config.providers</code> property. 
+  </p>
+  
+  <h4 class="anchor-heading">
+    <a id="using_providers" class="anchor-link"></a>
+    <a href="#using_providers">Using Configuration Providers</a>
+  </h4>
+  
+  <p>All configuration providers share a common way of passing parameters.</p>
+  <p>
+    You specify a comma-separated list of aliases and the fully-specified class that implements the configuration provider:
+  </p>
+  <pre><code>config.providers=provider1,provider2
+config.providers.provider1.class=com.example.Provider1
+config.providers.provider2.class=com.example.Provider2</code></pre>
+  <p>
+    Each provider can have its own set of parameters, which are passed in a specific format.
+  </p>
+  <p>
+    Pass additional parameters to each provider as <code>config.providers.&lt;provider_alias&gt;.param.&lt;name&gt;=&lt;value&gt;</code>:
+  </p>
+  <pre><code>config.providers.provider1.param.param1=value1
+config.providers.provider1.param.param2=value2</code></pre>
+  
+  <h4 class="anchor-heading">
+    <a id="config_provider" class="anchor-link"></a>
+    <a href="#config_provider">ConfigProvider</a>
+  </h4>
+  
+  <p>
+    The <code>ConfigProvider</code> is a base interface for all configuration providers. Custom implementations of this interface can retrieve configuration data from various sources. 
+    You can then package the implementation as a JAR file, add the JAR to your classpath, and reference the provider's class in your configuration.
+  </p>
+  
+  <p><b>Example custom provider configuration</b></p>
+  <pre><code>config.providers=customProvider
+config.providers.customProvider.class=com.example.customProvider
+config.providers.customProvider.param.param1=value1
+config.providers.customProvider.param.param2=value2</code></pre>
+  
+  <h4 class="anchor-heading">
+    <a id="directory_config_provider" class="anchor-link"></a>
+    <a href="#directory_config_provider">DirectoryConfigProvider</a>
```

Comment:
This provider has a configurable parameter. I think it would be nice to mention it here:
https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/config/provider/DirectoryConfigProvider.java#L48-L50